packages/server/src/users/database/UserSchema.ts

```diff
@@ -1,15 +1,80 @@
-import { Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Prop, raw, Schema, SchemaFactory } from '@nestjs/mongoose';
 import { Exclude, Expose } from 'class-transformer';
 import { ObjectId } from 'mongodb';
-import { Document } from 'mongoose';
+import mongoose, { Document } from 'mongoose';
 
 type UserDocument = User & Document<ObjectId>;
 
+const USER_CONSTANTS = Object.freeze({
+  CARD: Object.freeze({
+    NUMBER: {
+      REGEX:
+        /^(?:4[0-9]{12}(?:[0-9]{3})?)|(?:3[47][0-9]{13})|(?:5[1-5][0-9]{14})|(?:6(?:011|5[0-9][0-9])[0-9]{12})|(?:(?:2131|1800|35\d{3})\d{11})|(?:3(?:0[0-5]|[68][0-9])[0-9]{11})$/,
```

Comment:
Wygląda trochę strasznie. `[0-9]` można chyba zastąpić `\d` żeby było trochę krócej, chociaż nie wiem czy nie lepiej byłoby w RegExie ograniczyć to tylko do "od tylu do tylu cyfr, ewentualnie separator", a dalszą weryfikację robić już [kodem](https://en.wikipedia.org/wiki/Luhn_algorithm#Pseudocode_implementation), ale w sumie to raczej nie MVP.

Reply:
to znalazłem na necie, jak szukałem regexów do kart. To działa tak, że sprawdza początkowe numery kart, bo każda firma ma inne standary, że np Visa startuje od 4 zawsze. Sprawdziłem dla kilku przykładowych kart to niby działa, więc myślę, że możnaby zostawić, a jak się okaże problematyczne, to się wyrzuci